sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +385,352 @@ def transform(self, X):
 
         return X
+
+
+class MICEImputer(BaseEstimator, TransformerMixin):
+    """Basic implementation of MICE package from R.
+    This version assumes all of the features are Gaussian.
+
+    Parameters
+    ----------
+    missing_values : int or "NaN", optional (default="NaN")
+        The placeholder for the missing values. All occurrences of
+        missing_values will be imputed. For missing values encoded as
+        np.nan, use the string value "NaN".
+
+    imputation_order : str, optional (default="monotone")
+        The order in which the features will be imputed.
+
+        - "monotone" - From features with fewest missing values to most.
+        - "revmonotone" - From features with most missing values to fewest.
+        - "roman" - Left to right.
+        - "arabic" - Right to left.
+        - "random" - A random order for each round.
+
+    n_imputations : int, optional (defaults = 100)
+        Number of MICE rounds to perform the results of which will be
+        used in the final average.
+
+    n_burn_in : int, optional (default = 10)
+        Number of initial MICE rounds to perform the results of which
+        will not be returned.
+
+    n_nearest_features : int, optional (default = np.infty)
+        Number of other features to use to estimate the missing values of
+        the current features. Can provide significant speed-up
+        when the number of features is huge.
+
+    initial_fill_method : str, optional (default = "mean" )
+        Valid values: {"mean", "median", or "most_frequent"}.
+        Uses :class:sklearn.preprocessing.Imputer.
+
+    min_value : float (default = -np.infty)
+        Minimum possible imputed value.
+
+    max_value : float (default = np.infty)
+        Maximum possible imputed value.
+
+    verbose : boolean (default = False)
+        Whether to provide updates on progress.
+
+    random_state : int seed, RandomState instance (default = None)
+        The seed of the pseudo random number generator to use when
+        shuffling the data.
+
+    Notes
+    -----
+    Features which only contain missing values at `fit` are discarded
+        upon `transform`.
+
+    References
+    ----------
+    .. [1] `Stef van Buuren, Karin Groothuis-Oudshoorn (2011). "mice:
+        Multivariate Imputation by Chained Equations in R". Journal of
+        Statistical Software 45: 1-67.
+        <https://www.jstatsoft.org/article/view/v045i03>`_
+    """
+
+    def __init__(
+            self,
+            missing_values='NaN',
+            imputation_order='monotone',
+            n_imputations=100,
+            n_burn_in=10,
+            n_nearest_features=np.infty,
+            initial_fill_method="mean",
+            min_value=-np.infty,
+            max_value=np.infty,
+            verbose=False,
+            random_state=None):
+        from ..linear_model import BayesianRidge  # avoiding circular import
+        self.model = BayesianRidge()  # TODO: need to move to fit_transform?
```

Comment:
I would change `model` by `estimator`. You need to make this initialization in `fit`
If you wish to pass it as a parameter, you can pass `None`. Basically, you would have those 2 cases.

### If `estimator` is a parameter:

```python

def __init__(self, estimator=None, ...):
    self.estimator = estimator

def fit(self, ...):
    if self.estimator is None:
        from ..linear_model import BayesianRidge
        self.estimator_ = BayesianRidge(random_state=self.random_state)
    else:
        self.estimator_ = clone(self.estimator)
```

### if `estimator` is not a parameter in init

```python
def fit(self, ...):
        from ..linear_model import BayesianRidge
        self.estimator_ = BayesianRidge(random_state=self.random_state)
```

Reply:
Great, thanks. I wasn't sure how to do this properly.